src/main/scala/com/udemy/statistics/algorithm/Sampling.scala

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2018 Udemy, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package com.udemy.statistics.algorithm
+
+import scala.Numeric.Implicits._
+import scala.collection.parallel.immutable.ParVector
+import scala.util.Random
+import scalaz.NonEmptyList
+import scalaz.Scalaz._
+
+object Sampling {
+
+  def randomDraw[T:Numeric](x: NonEmptyList[T]): T =
+    x.index(Random.nextInt(x.length)).get
+
+  def drawSample[T: Numeric](x: Seq[T], size: Int): Seq[T] = {
```

Comment:
how might a function that samples without replacement be related to this function? should we add a boolean argument like in R?

Reply:
I considered that actually. I started adding the argument and then realized that I would then have to implement the sampling without replacement functionality and didn't want to do it in this PR. My thought is when we add that, we can make with replacement the default so as to maintain backward compatibility.

Reply:
Okay, added support for sampling with replacement. Also, renamed per our offline conversation.